keps/sig-autoscaling/20181106-in-place-update-of-pod-resources.md

```diff
@@ -105,126 +109,177 @@ Other identified non-goals are:
 
 ### API Changes
 
-PodSpec becomes mutable with regards to resources and limits.
+PodSpec becomes mutable with regards to Container resources requests and limits.
 Additionally, PodSpec becomes a Pod subresource to allow fine-grained access control.
 
-PodStatus is extended with information about actually allocated resources.
+PodStatus is extended with information about actually allocated Container resources.
 
 Thanks to the above:
 * PodSpec.Container.ResourceRequirements becomes purely a declaration,
   denoting **desired** state of the Pod,
 * PodStatus.ContainerStatus.ResourceAllocated (new object) denotes **actual**
   state of the Pod resources.
 
-To distinguish between possible states of the Pod resources,
-a new PodCondition InPlaceResize is added, with the following states:
+To distinguish between possible states of a Pod resource update,
+a new PodCondition named ResourceResizeRequired is added, with the following states:
 * (empty) - the default value; resource update awaits reconciliation
-  (if ResourceRequirements differs from ResourceAllocated),
+  if ResourceRequirements differs from ResourceAllocated,
+* Requested - Scheduler determined in-place resource resizing is possible, and
+  requested Kubelet to update Pod's resource allocations and limits,
 * Awaiting - awaiting resources to be freed (e.g. via pre-emption),
-* Failed - resource update could not have been performed in-place
+* Failed - resource resizing could not have been performed in-place
   but might be possible if some conditions change,
 * Rejected - resource update was rejected by any of the components involved.
 
 To provide some fine-grained control to the user,
-PodSpec.Container.ResourceRequirements is extended with ResizingPolicy flag,
-available per each resource request (CPU, memory) :
-* InPlace - the default value; allow in-place resize of the Container,
-* RestartContainer - restart the Container to apply new resource values
-  (e.g. Java process needs to change its Xmx flag),
-* RestartPod - restart whole Pod to apply new resource values
+PodSpec.Container.ResourceRequirements is extended with ResizePolicy flag
+for each resource type (CPU, memory) :
+* Update - the default value; resize the Container without restart,
```

Comment:
I would use a different name for this option, something like `NoRestart` or `NonDisruptiveResize`. "Update" is very generic. It gives the impression that the other options do not "update" as opposed to this one that does. There is no information about restart behavior either.

Reply:
'NoRestart' sounds much better, I'll update it.